lib/IRGen/IRGen.cpp

```diff
@@ -244,6 +261,18 @@ void swift::performLLVMOptimizations(const IRGenOptions &Opts,
 
   PassBuilder PB(TargetMachine, PTO, PGOOpt, &PIC);
 
+  // Attempt to load pass plugins and register their callbacks with PB.
+  for (const auto &PluginFile : Opts.LLVMPassPlugins) {
+    Expected<PassPlugin> PassPlugin = PassPlugin::Load(PluginFile);
+    if (PassPlugin) {
+      PassPlugin->registerPassBuilderCallbacks(PB);
+    } else {
+      diagnoseSync(Diags, DiagMutex, SourceLoc(),
+                   diag::unable_to_load_pass_plugin, PluginFile,
+                   toString(PassPlugin.takeError()));
```

Comment:
I wonder if the diagnostic handling should remain in `performLLVM()`. Passing `Diags` and `DiagMutex` to `performLLVMOptimizations()` seems a bit unfortunate. We could instead just return the `llvm::Error` from here and let the callers take care of it. That would change the return type of the function. A lot of code in *include/swift* is using `llvm::Error` already, but I am not sure what the conventions are in Swift. @lhames What do you think?